designs/2020-rule-tester-only/README.md

```diff
@@ -0,0 +1,171 @@
+- Repo: eslint/eslint
+- Start Date: 2020-12-27
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Brandon Mills ([@btmills](https://github.com/btmills))
+
+# `RuleTester` test isolation with `only`
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+`RuleTester` currently lacks a built-in way for developers to isolate individual tests during development.
+Temporarily deleting or commenting out the other tests is tedious.
+This adds an optional `only` property to run individual tests in isolation.
+
+## Motivation
+
+<!-- Why are we doing this? What uses does it support? What is the expected
+outcome? -->
+
+When developers are working on a rule, we can run its tests with `npm run test:cli tests/lib/rules/my-rule.js`.
+Debugging sometimes requires focusing on a particular test and running it in isolation to avoid unrelated noise from other tests.
+
+Tools like Mocha already offer `it.only()` for low-friction test isolation.
+Developers may already be familiar with that API.
+Exposing a similar API via `RuleTester` would improve the ergonomics of debugging tests.
+This API is beneficial to any approach from `console.log("HERE");` to interactive debugging with breakpoints.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corners, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+Add an optional boolean `only` property to `RuleTester`'s `ValidTestCase` and `InvalidTestCase` types.
+Add `only` to the other `RuleTester`-only properties in the `RuleTesterParameters` array to exclude it from configuration schema validation.
+
+To allow using `RuleTester` with a custom test framework other than Mocha, parallel `RuleTester`'s existing `describe` and `it` implementations for `itOnly`:
+
+1. Declare an `IT_ONLY` `Symbol`, set it as a static property on `RuleTester`, and initialize it to `null`.
+1. Add an `itOnly` `set` accessor that sets `RuleTester[IT_ONLY]`.
+1. Add an `itOnly` `get` accessor.
+   1. If `RuleTester[IT_ONLY]` is set, return it.
+   2. If global `it` and `it.only` are functions, return `it.only`. *TODO: Is `this` necessary? Could return a wrapper instead.*
```

Comment:
I don't think we have a use for `this` in our tests, so should be fine to return `it.only` directly.

Reply:
We could return `Function.bind.call(it.only, it)` for the case that `only` expects `this` to be `it`.

Reply:
That works.

Reply:
That's easy enough to do just in case. Updated.